lib/cn/csched_sp3.c

```diff
@@ -1389,8 +1387,7 @@ sp3_work_progress(struct cn_compaction_work *w)
     struct cn_merge_stats ms;
     uint progress;
 
-    progress = (w->cw_stats.ms_keys_in * 100) /
-        clamp_t(uint64_t, est->cwe_keys, 1, est->cwe_keys);
+    progress = (w->cw_stats.ms_keys_in * 100) / max_t(uint64_t, 1, est->cwe_keys);
```

Comment:
Can keys_in every be greater than cwe_keys?

Reply:
If msg_type is final, wouldn't progress be 100% ?   Maybe just set it to 100 in that case.

For the other case, you might just hard code to 0 or 100% if cwe_keys==0.

Reply:
Doesn't seem so. cwe_keys seems to be calculated as all keys in all kvsets, where as ms_keys_in a running count of all keys that have entered the compaction operation